core/src/main/scala/chisel3/Data.scala

```diff
@@ -826,7 +859,12 @@ object Data {
     *  add to Data we also want on Connectable, so an implicit conversion makes the most sense
     *  so the ScalaDoc can be shared.
     */
-  implicit def toConnectableDefault[T <: Data](d: T): Connectable[T] = Connectable.apply(d)
+  implicit def toConnectableDefault[T <: Data](d: T): Connectable[T] = Connectable.apply(
+    d,
+    d.connectableWaiveSelection,
+    d.connectableSqueezeSelection,
+    d.connectableExcludeSelection
+  )
```

Comment:
It seems we're doing a sort of mix of typeclasses ("static"* dispatch) and inheritance (dynamic dispatch). What this means is that I don't think this will compose the way a user expects. If you override these protected methods in a Bundle and then instantiate a Bundle as a child in a parent Bundle, I don't think these methods will be called for the child. To see if I'm right, try taking `9.a.1` and putting `MyBundle` inside of another `Bundle` and check what happens upon the connectable operator.

If I am right, then I think we need to consider an alternative implementation. Rather than trying to bind this information statically, we need to check these fields of any child Bundles dynamically in the implementation of the Connectable operators, i.e. in [doConnection](https://github.com/chipsalliance/chisel/blob/84c0fbfba6eecc2e53808d5638135036376c9483/core/src/main/scala/chisel3/connectable/Connection.scala#L131).


*Typeclasses in Scala are technically implemented with inheritance and thus dynamic dispatch, but the point is that the actual method that will ultimately be called is chosen (and thus analyzable) statically.

Reply:
Ah that's a good point, let me confirm but I think you're right.

Reply:
Yep, you're exactly right, this only applies to the top-level data when the implicit conversion happens.